src/components/DataTable/components/SbDataTableBody.js

```diff
@@ -16,7 +16,7 @@ export const SbDataTableBodyRow = {
       type: Array,
     },
     row: {
-      type: Object,
+      type: Object | Array,
```

Comment:
I would like to recommend to use the array notation, [just for standardization](https://vuejs.org/v2/guide/components-props.html#Prop-Validation)